pkg/gatewayserver/io/io.go

```diff
@@ -233,6 +235,16 @@ func (c *Connection) HandleStatus(status *ttnpb.GatewayStatus) error {
 	case c.statusCh <- status:
 		c.lastStatus.Store(status)
 		atomic.StoreInt64(&c.lastStatusTime, time.Now().UnixNano())
+
+		antennaLocations := status.GetAntennaLocations()
+		if len(antennaLocations) > 0 && c.gateway.UpdateLocationFromStatus {
+			// TODO: Handle multiple antenna locations (https://github.com/TheThingsNetwork/lorawan-stack/issues/2006).
+			select {
+			case c.locCh <- antennaLocations[0]:
+			default:
+				log.FromContext(c.ctx).Warnf("Drop gateway location")
+			}
```

Comment:
Why are you treating locations like traffic, i.e. storing them in a buffer, but then draining the buffer to get the latest location?

We are already storing state in `c .lastStatus`. Why not send a signal over a channel that the last status updated? That can be a buffered channel of size 1. You don't need to warn here if that channel is full; the signal is ready in the channel, and the receiver can read `c.lastStatus` and get the latest location from there. That also decouples functionality; now there's location stuff here _and_ in the goroutine.

Reply:
Updated to use a signal channel instead.